File: DMOJ/An Animal Contest/Contest 5/BambooCookies.py
# An Animal Contest 5 P1 - Bamboo Cookies

n = int(input())
e, o = 0, 0
for i in map(int, input().split()):
    if i % 2:
        e += 1
    else:
        o += 1

print(e // 2 + o // 2)

File: DMOJ/An Animal Contest/Contest 5/BambooCookies.go
// An Animal Contest 5 P1 - Bamboo Cookies

package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
)

var _tk []string
var _iop int = 0

func su() int {
	r, _ := strconv.Atoi(_tk[_iop])
	_iop++
	return r
}
func main() {
	bytes, _ := ioutil.ReadAll(os.Stdin)
	_tk = strings.Fields(string(bytes))

	n := su()
	e := 0
	for i := 0; i < n; i++ {
		e += su() % 2
	}
	fmt.Println(e/2 + (n-e)/2)
}
